src/main/java/bio/terra/janitor/service/cleanup/FlightManager.java

```diff
@@ -0,0 +1,113 @@
+package bio.terra.janitor.service.cleanup;
+
+import bio.terra.janitor.db.CleanupFlightState;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.janitor.db.TrackedResource;
+import bio.terra.stairway.Stairway;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightNotFoundException;
+import bio.terra.stairway.exception.StairwayException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages the Flights to clean up tracked resources. It queries {@link JanitorDao} and {@link
+ * Stairway} to hand off work between the two.
+ *
+ * <p>This class is meant to be run by only the primary Janitor instance, not by the secondaries.
+ *
+ * <p>The handoff is done by this class and tracked in the cleanup_flight table and Stairway's
+ * database.
+ */
+public class FlightManager {
+  private Logger logger = LoggerFactory.getLogger(FlightManager.class);
+
+  /**
+   * A limit on how many flights to recover at startup. This is to prevent blowing out the service
+   * at start up, but if we have too many flights to recover, they will not all be recovered
+   * immediately.
+   */
+  private static final int RECOVERY_LIMIT = 1000;
+
+  private final Stairway stairway;
```

Comment:
Can we use dependency injection for this class? 
i.e., autowire the three parameters and make this component?

Reply:
We don't have a FlightSubmissionFactory to wire for this class yet. And I don't actually want this class to be available from anywhere in the applciation, so not going to autowire this class itself.

Reply:
(1) Just to confirm what I understand: For now we don't have FlightSubmissionFactory to wire so we are not injecting 
staritway, janitorDaom, submissionFactory yet. Instead, we will manually creates/pass them from upsteam services? 
But once we have FlightSubmissionFactory to wire, we will switch to inject them,
(2) We will not make the class a compoment? 
(3) Also, make the class injectable vs controling the usage of the class are different things. We can always have some ways to control the access

Reply:
1) Yes. See https://github.com/DataBiosphere/terra-resource-janitor/pull/24
2) I don't think we need to, so I don't know why we would.